cmd/wire/main.go

```diff
@@ -118,20 +124,24 @@ func generate(pkgs ...string) error {
 //
 // Given one or more packages, diff will generate the content for the
 // wire_gen.go file, and output the diff against the existing file.
-func diff(pkgs ...string) error {
+//
+// Similar to the diff command, it returns 0 if no diff, 1 if different, 2
+// plus an error if trouble.
+func diff(pkgs ...string) (int, error) {
```

Comment:
Optional: I wonder if this would be simpler to reason about if it returned `(bool, error)` and then `main()` translated from the bool to an exit code. The magic `2` has to be copied a fair amount and it seems necessary to code defensively above to ensure the invariant is upheld.

Reply:
Nice. That resolves my concern.